### DISCUSSION TITLE
Phalén Calidi XP Smart Pool Heatpump

### DISCUSSION THREAD

**Discussion by kimhey** [make-all/tuya-local#988] [General]
Great! The Pool heat pump Calidi XP by swedish supplier Phalén works perfect with same configuration as the Fairland Heatpump. I can now control temperature and see actual temp as well as see diagnostics of power level (%) and waterflow status. Thanks alot!

**Comment by make-all** [comment]
Do you have a product id for this device?  Soon I will be making the modifications to discover devices by product id, so it will be good to have things ready (the current matching by dps will still be a backup, but is less reliable with many devices now not sending all their dps when queried for status).

**Reply by kimhey** [reply]
Hi, sorry for super slow response. If you are still interested in adding this specific device I'll try to give you the details I could find:
Item number: 14983341
Model number: XP15
Model name: Calidi XP

Didn't find any specific tag for "product id" but let me know if you know where to look.

**Reply by make-all** [reply]
product id will be in the tuya cloud info, or in the result of `python -m tinytuya scan`

If you can't find it, no problem. When we start to use it, unrecognised product ids will get logged, so you can get it then.

**Reply by kimhey** [reply]
Ah, I see. Then this should be the info you are looking for:

"name": "Inverter Pool Heat Pump",
      "product_id": "qrlLaHWwIsZsV31f",
      "product_name": "OEM-单温Inverter Pool Heat Pump"

**Reply by kimhey** [reply]
It showed up as "Fairland InverterPlus pool Heatpump" in HA, not sure why. Maybe they are sharing same wifi module or something?
